### PR TITLE
fix(discover) Fix negated issue conditions

### PR DESCRIPTION
**Pull request by markstory**
The issue condition should support negation like other conditions. I've changed the semantics a bit on how the issue filter works as it no longer goes into the `group_ids` filter key, but from what I saw snuba can still promote the new condition into a prewhere.

### CODE REVIEW

File: tests/snuba/api/endpoints/test_organization_events_v2.py [hunk 1]
```diff
@@ -1623,6 +1623,49 @@ def test_issue_in_search_and_columns(self):
...
+                data={
+                    "field": ["title"],
+                    "query": "!issue:{}".format(event1.group.qualified_short_id),
+                },
+            )
+
+            assert response.status_code == 200, response.content
+            data = response.data["data"]
+            assert len(data) == 1
+            assert data[0]["title"] == event1.title
```
Comment (dashed) [inline tests/snuba/api/endpoints/test_organization_events_v2.py:1667]: @markstory this should be the title of the 2nd event.

File: src/sentry/api/event_search.py [hunk 2]
```diff
@@ -760,6 +760,14 @@ def convert_search_filter_to_snuba_query(search_filter, key=None):
                 )
             )
         return [name, search_filter.operator, internal_value]
+    elif name == "issue.id":
+        # Handle "has" queries
+        if search_filter.value.raw_value == "":
+            return [["isNull", [name]], search_filter.operator, 1]
```
Comment (evanh) [inline src/sentry/api/event_search.py:766]: If group_id is not nullable, why do we do this for has? Can we just hardcode '1' or something to take some load off clickhouse?

File: tests/sentry/api/test_event_search.py [hunk 3]
```diff
@@ -1185,12 +1185,30 @@ def test_issue_id_filter(self):
...
+        assert _filter.conditions == [["issue.id", "=", group.id]]
+        assert _filter.filter_keys == {}
+        assert _filter.group_ids == []
+
+    def test_negated_issue_filter(self):
+        group = self.create_group(project=self.project)
+        _filter = get_filter(
+            "!issue:{}".format(group.qualified_short_id), {"organization_id": self.organization.id}
+        )
+        assert _filter.conditions == [["issue.id", "!=", 2]]
```
Comment (evanh) [inline tests/sentry/api/test_event_search.py:1208]:
```suggestion
        assert _filter.conditions == [["issue.id", "!=", group.id]]
```

File: tests/snuba/api/endpoints/test_organization_events_v2.py (hunk 1 above)
Comment (markstory) [reply tests/snuba/api/endpoints/test_organization_events_v2.py:1667]: Whoops.

File: src/sentry/api/event_search.py (hunk 2 above)
Comment (markstory) [reply src/sentry/api/event_search.py:766]: We could hardcode the group_id condition to 0 and skip the isNull call.

File: tests/snuba/api/endpoints/test_organization_events_v2.py (hunk 1 above)
Comment (markstory) [reply tests/snuba/api/endpoints/test_organization_events_v2.py:1667]: This test was passing by luck as the two events have the same title :blush: I'll get that fixed.

**Review by dashed — APPROVED** [review]
Looks good to me 👍

**Pull request merged**